### PR TITLE
feat(column): Added inner/outer padding

### LINKED ISSUE #2968: Column chart column width
In our use case, we have a column chart that is set for a consistent size with the other charts on our page, but due to the nature of the data, there aren't many columns across the x-axis, and thus the width of the columns is too much and looks inconsistent with our multi-series data.  Could there be a way to manipulate the width of the columns on the chart?

### PR DESCRIPTION
**Pull request by jbrundage**
Fixes #2968

Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

### CODE REVIEW

**Comment by jbrundage** [comment]
image

File: packages/chart/src/Axis.ts [hunk 1]
```diff
@@ -698,3 +703,4 @@ Axis.prototype.publish("labelRotation", 33, "number", "Angle of rotation for tic
 Axis.prototype.publish("shrinkToFit", "both", "set", "shrinkToFit", ["none", "low", "high", "both"]); // TODO: What does this control?
 Axis.prototype.publish("extend", 5, "number", "Extend the axis range by this % beyond what is needed to display the data (disabled when type is 'ordinal')", null, { optional: true, disable: (w: any) => w.type() === "ordinal" });
 Axis.prototype.publish("hidden", false, "boolean", "Hides axis when 'true'");
+Axis.prototype.publish("scalePadding", 0.1, "number", "Determines the ratio of the range that is reserved for blank space before the first band, after the last band and between bands");
```
Comment (GordonSmith) [inline packages/chart/src/Axis.ts]: rename to "ordinalPaddingOuter"

File: packages/chart/src/Column.ts [hunk 2]
```diff
@@ -258,13 +257,16 @@ export interface Column {
...
+    paddingInner(): number;
+    paddingInner(_: number): this;
 }
 
 Column.prototype.publish("paletteID", "default", "set", "Color palette for this widget", () => Column.prototype._palette.switch(), { tags: ["Basic", "Shared"] });
 Column.prototype.publish("useClonedPalette", false, "boolean", "Enable or disable using a cloned palette", null, { tags: ["Intermediate", "Shared"] });
 Column.prototype.publish("showValue", false, "boolean", "Show Value in column");
 Column.prototype.publish("valueCentered", false, "boolean", "Show Value in center of column");
 Column.prototype.publish("valueAnchor", "middle", "set", "text-anchor for shown value text", ["start", "middle", "end"]);
+Column.prototype.publish("paddingInner", 0, "number", "Determines the ratio of the range that is reserved for blank space between bands");
```
Comment (GordonSmith) [inline packages/chart/src/Column.ts]: Rename to ordinalPaddingInner

File: packages/chart/src/Column.ts [hunk 3]
```diff
@@ -74,12 +71,13 @@ export class Column extends XYAxis {
...
         switch (host.xAxisType()) {
             case "ordinal":
+                host.paddingOuter(this.paddingOuter());
                 dataLen = host.bandwidth();
                 offset = -dataLen / 2;
                 break;
             case "linear":
             case "time":
-                dataLen = Math.max(Math.abs(host.dataPos(2) - host.dataPos(1)) * (100 - this._linearGap) / 100, dataLen);
+                dataLen = Math.max(Math.abs(host.dataPos(2) - host.dataPos(1)) * this.paddingOuter(), dataLen);
```
Comment (jbrundage) [inline packages/chart/src/Column.ts]: @GordonSmith renaming `scalePadding` to `ordinalPaddingOuter` would affect these liner/time paddings... should I create another publish parameter for these? or...

**Comment by jbrundage** [comment]
@GordonSmith on this PR thread TravisCI shows "Pending — The Travis CI build is in progress", but clicking "Details" shows that the test passed 23 hours ago?

**Comment by GordonSmith** [comment]
Closing in favor of:  https://github.com/hpcc-systems/Visualization/pull/2993

**Pull request closed without merging**